pkg/odo/genericclioptions/util.go

```diff
@@ -34,3 +36,16 @@ func ApplyIgnore(ignores *[]string, sourcePath string) (err error) {
 
 	return nil
 }
+
+// WarnIfDefaultNamespace warns when user tries to run `odo dev` or `odo deploy` in the default namespace
+func WarnIfDefaultNamespace(namespace string, kubeClient kclient.ClientInterface) {
+	if namespace == "default" {
```

Comment:
nit: wondering if we could use the `NamespaceDefault` constant available in https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#pkg-constants